### PR TITLE
MSI-2657: Eliminate button 'Ship' in orders placed with Store pickup method.

### PR DESCRIPTION
**Pull request by nmalevanec**
### Fixed Issues (if relevant)

1. https://github.com/magento/inventory/issues/2657: Eliminate button 'Ship' in orders placed with Store pickup method

### Manual testing scenarios (*)

1. Create order with store pickup shipping method.
2. Navigate to admin order page.
3. Observe 'Ship' button is absent on order page.

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/code/Magento/InventoryInStorePickupAdminUi/Plugin/Sales/Block/Adminhtml/Order/ProcessShipButtonPlugin.php [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        View $subject,
+        $proceed,
+        $buttonId,
+        $data,
+        $level = 0,
+        $sortOrder = 0,
+        $region = 'toolbar'
+    ): View {
+        if ($buttonId === 'order_ship') {
+            if ($subject->getOrder()->getShippingMethod() === 'in_store_pickup') {
```
Comment (swnsma) [inline app/code/Magento/InventoryInStorePickupAdminUi/Plugin/Sales/Block/Adminhtml/Order/ProcessShipButtonPlugin.php]: May we use constant `\Magento\InventoryInStorePickupShippingApi\Model\Carrier\InStorePickup::DELIVERY_METHOD` instead?

**Review by swnsma — CHANGES_REQUESTED** [review]

File: app/code/Magento/InventoryInStorePickupAdminUi/Plugin/Sales/Block/Adminhtml/Order/ProcessShipButtonPlugin.php (hunk 1 above)
Comment (nmalevanec) [reply app/code/Magento/InventoryInStorePickupAdminUi/Plugin/Sales/Block/Adminhtml/Order/ProcessShipButtonPlugin.php]: Done.

**Comment by sidolov** [comment]
@magento run Functional Tests build

**Comment by sidolov** [comment]
@magento run Functional Tests

**Pull request merged**